File: 30days/day26-nested-logic/Solution.js
// PROBLEM: https://www.hackerrank.com/challenges/30-nested-logic/problem

const toDateObject = (line) => {
    const [day, month, year] = line.split(' ').map(Number);
    return { year, month, day};
}

function processData(input) {
    //Enter your code here
    const returned = toDateObject(input.split('\n')[0]);
    const expected = toDateObject(input.split('\n')[1]);
    let fine = 0;

    if (expected.year === returned.year && expected.month === returned.month) {
        fine = (returned.day - expected.day) * 15;
    } else if (expected.year === returned.year && expected.month < returned.month) {
        fine = (returned.month - expected.month) * 500;
    } else if (expected.year < returned.year) {
        fine = 10000;
    }

    console.log(fine);
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: 30days/day26-nested-logic/Solution.cs
// PROBLEM: https://www.hackerrank.com/challenges/30-nested-logic/problem

using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        DateTime returned = ToDateTime(Console.ReadLine());
        DateTime expected = ToDateTime(Console.ReadLine());
        int fine = 0;

        if (expected.Year == returned.Year && expected.Month == returned.Month) {
            fine = (returned.Day - expected.Day) * 15;
        } else if (expected.Year == returned.Year && expected.Month < returned.Month) {
            fine = (returned.Month - expected.Month) * 500;
        } else if (expected.Year < returned.Year) {
            fine = 10000;
        }

        Console.WriteLine(fine);
    }

    static DateTime ToDateTime(string line) {
        return new DateTime(Convert.ToInt32(line.Split(' ')[2]), Convert.ToInt32(line.Split(' ')[1]), Convert.ToInt32(line.Split(' ')[0]));
    }
}